### PR TITLE
Upgrade to go 1.14.1

### PR DESCRIPTION
**Pull request by fjordan**
Alternative to https://github.com/Shopify/ghostferry/pull/125. 

Upgrades to go **1.14.1** and introduces **go modules**.

The previous version was 1.10, and was still using `dep`. 

With the upgrade, this moves to go modules, but can still make the `vendor` dir if we'd like by using `go build -mod=vendor`.  The `vendor` directory was persisted using `go mod vendor` after running `go mod download` and `go mod tidy`.

/cc @Shopify/pods

### CODE REVIEW

**Comment by shuhaowu** [comment]
Did you update the siddontang/go-mysql dependency? It's usually worth looking at a diff and a commit log so we can see if they changed something that may be important to us.

**Comment by shuhaowu** [comment]
This is great work by the way. Thanks a tonne.

File: config.go [hunk 1]
```diff
@@ -64,14 +65,14 @@ type DatabaseConfig struct {
...
-		Params:    c.Params,
-
-		MultiStatements: true,
+		User:                 c.User,
+		Passwd:               c.Pass,
+		Net:                  "tcp",
+		Addr:                 fmt.Sprintf("%s:%d", c.Host, c.Port),
+		Collation:            c.Collation,
+		Params:               c.Params,
+		AllowNativePasswords: true,
```
Comment (fjordan) [inline config.go:74]: This is used for the tests

File: Makefile [hunk 2]
```diff
@@ -34,7 +34,7 @@ DEB_TARGET       = $(BUILD_DIR)/$(PROJECT_BIN)_$(VERSION_STR).deb
 
 $(PROJECTS): $(GOBIN)
 	$(eval proj := $@)
-	go build -i -ldflags "$(LDFLAGS)" -o $(BIN_TARGET) $(PROJECT_PKG)
+	go build -mod=vendor -i -ldflags "$(LDFLAGS)" -o $(BIN_TARGET) $(PROJECT_PKG)
```
Comment (fjordan) [inline Makefile:37]: This is to ensure we're building with the dependencies in the `vendor/` directory

**Pull request merged**